### PR TITLE
Add wallpaperFillMode configuration option

### PR DESCRIPTION
**Pull request by bemyak**
This PR adds a new `wallpaperFillMode` configuration option to set the wallpaper scaling method.

### CODE REVIEW

File: lib/wallpapers.nix [hunk 1]
```diff
@@ -27,4 +27,16 @@
...
+  #  https://invent.kde.org/plasma/kdeplasma-addons/-/blob/bc53d651cf60709396c9229f8c582ec8a9d2ee53/applets/mediaframe/package/contents/ui/ConfigGeneral.qml#L148-170
+  wallpaperFillModeTypes = {
+    "Stretch" = 0; # a.k.a. Scaled
+    "PreserveAspectFit" = 1; # a.k.a. Scaled Keep Proportions
+    "PreserveAspectCrop" = 2; # a.k.a. Scaled And Cropped
+    "Tile" = 3;
+    "TileVertically" = 4;
+    "TileHorizontally" = 5;
+    "Pad" = 6; # a.k.a. Centered
+  };
```
Comment (HeitorAugustoLN) [inline lib/wallpapers.nix]:
```suggestion
  wallpaperFillModeTypes = {
    "stretch" = 0; # a.k.a. Scaled
    "preserveAspectFit" = 1; # a.k.a. Scaled Keep Proportions
    "preserveAspectCrop" = 2; # a.k.a. Scaled And Cropped
    "tile" = 3;
    "tileVertically" = 4;
    "tileHorizontally" = 5;
    "pad" = 6; # a.k.a. Centered
  };
```

Settings values are should be named in camelCase for consistency

File: modules/workspace.nix [hunk 2]
```diff
@@ -164,6 +164,16 @@ in
...
 
+    wallpaperFillMode = with lib.types; lib.mkOption {
+      type = enum (lib.attrNames wallpaperFillModeTypes);
+      default = "Stretch";
+      example = "PreserveAspectFit";
+      description = ''
+        Defines how the wallpaper should be displayed on the screen.
+        Applies only to wallpaperPictureOfTheDay or wallpaperSlideShow.
+      '';
+    };
```
Comment (HeitorAugustoLN) [inline modules/workspace.nix]:
```suggestion
    wallpaperFillMode = lib.mkOption {
      type = with lib.types; nullOr (enum (builtins.attrNames wallpaperFillModeTypes));
      default = null;
      example = "stretch";
      description = ''
        Defines how the wallpaper should be displayed on the screen.
        Applies only to wallpaperPictureOfTheDay or wallpaperSlideShow.
      '';
      apply = value: if value == null then null else (builtins.toString wallpaperFillModeTypes.${value});
    };
```

File: modules/panels.nix [hunk 3]
```diff
@@ -211,6 +213,7 @@ in
                   "\"" + (builtins.toString path) + "\"" else
                   "[" + (builtins.concatStringsSep "," (map (s: "\"" + s + "\"") path)) + "]"});
                 desktop.writeConfig("SlideInterval", "${builtins.toString cfg.workspace.wallpaperSlideShow.interval}");
+                desktop.writeConfig("FillMode", "${builtins.toString wallpaperFillModeTypes.${cfg.workspace.wallpaperFillMode}}");
```
Comment (HeitorAugustoLN) [inline modules/panels.nix]:
```suggestion
                ${lib.optionalString (cfg.workspace.wallpaperFillMode != null) ''desktop.writeConfig("FillMode", "${builtins.toString cfg.workspace.wallpaperFillMode}");''}
```

File: modules/panels.nix [hunk 4]
```diff
@@ -221,6 +224,7 @@ in
                 desktop.currentConfigGroup = ["Wallpaper", "org.kde.potd", "General"];
                 desktop.writeConfig("Provider", "${cfg.workspace.wallpaperPictureOfTheDay.provider}");
                 desktop.writeConfig("UpdateOverMeteredConnection", "${if (cfg.workspace.wallpaperPictureOfTheDay.updateOverMeteredConnection) then "1" else "0"}");
+                desktop.writeConfig("FillMode", "${builtins.toString wallpaperFillModeTypes.${cfg.workspace.wallpaperFillMode}}");
```
Comment (HeitorAugustoLN) [inline modules/panels.nix]:
```suggestion
                ${lib.optionalString (cfg.workspace.wallpaperFillMode != null) ''desktop.writeConfig("FillMode", "${builtins.toString cfg.workspace.wallpaperFillMode}");''}
```

**Comment by HeitorAugustoLN** [comment]
Besides that it should work fine. I will test it after those changes.

**Comment by bemyak** [comment]
Thanks for the review!

**Review by HeitorAugustoLN — APPROVED** [review]

**Comment by HeitorAugustoLN** [comment]
It works flawlessly. Thanks for your contribution!

**Comment by bemyak** [comment]
I learned a lot from your suggestions and comments :blush:

**Pull request merged**